src/pages/sicp/Sicp.tsx

```diff
@@ -89,13 +94,31 @@ const Sicp: React.FC = () => {
 
   // Handle loading of latest viewed section and fetch json data
   React.useEffect(() => {
+    const valid_langs = ['en', 'zh_CN'];
+
+    if ((section && valid_langs.includes(section)) || param_lang) {
+      const plang = param_lang ? param_lang : section ? section : SICP_DEF_TB_LANG;
+      if (!valid_langs.includes(plang)) {
+        setLang(SICP_DEF_TB_LANG);
+        setSicpLangLocalStorage(SICP_DEF_TB_LANG);
+      } else {
+        setLang(plang);
+        setSicpLangLocalStorage(plang);
+      }
+      if (section && valid_langs.includes(section)) {
+        navigate(`/sicpjs/${SICP_INDEX}`, { replace: true });
+      } else {
+        navigate(`/sicpjs/${section}`, { replace: true });
+      }
+      return;
+    }
     if (!section) {
       /**
        * Handles rerouting to the latest viewed section when clicking from
        * the main application navbar. Navigate replace logic is used to allow the
        * user to still use the browser back button to navigate the app.
        */
-      navigate(`/sicpjs/${readSicpSectionLocalStorage()}`, { replace: true });
+      navigate(path.join('sicpjs', readSicpSectionLocalStorage()), { replace: true });
```

Comment:
Why was this changed? `path` is a node standard library built-in, so doesn't exist in the browser without a polyfill